utils/dd/dd_list.c

```diff
@@ -206,38 +206,55 @@ int main(int argc, char *argv[])
 
     }
 
-    if (!directory || !versions.kernel || !versions.anaconda) {
+    if (!path || !versions.kernel || !versions.anaconda) {
         show_help();
         rc = 1;
         goto cleanup;
     }
 
-    if (verbose) {
-        printf("Listing DD dir %s\n", directory);
-    }
-
     init_rpm();
 
-    char *globpattern;
-    checked_asprintf(&globpattern, "%s/*.rpm", directory);
+    /* determine if path is rpm filename or directory
+     * (paths ending with "/.rpm" will be still treated as directories) */
+    if((strlen(path) > 5) && (!strcmp(path + strlen(path) - 4, ".rpm")) &&
+            (*(path + strlen(path) - 5) != '/')) {
```

Comment:
It would make sense to compute the length of the ``path`` string once and then use the computed value.